src/atoms/Switch/Switch.native.js

```diff
@@ -0,0 +1,315 @@
+        duration: SWITCH_STATE_CONFIGS.KNOB_SCALE_DURATION,
+      }).start();
+    } else {
+      Animated.timing(rightSpace.current, {
+        toValue: SWITCH_STATE_CONFIGS.OFF_ACTIVE_RIGHT_SPACE,
+        duration: SWITCH_STATE_CONFIGS.KNOB_SCALE_DURATION,
+      }).start();
+    }
+    setActive(true);
+  };
+
+  const onPressOut = () => {
+    setActive(false);
+    if (isPropDefined(on)) {
+      if (toggleState) {
+        Animated.timing(leftSpace.current, {
+          toValue: SWITCH_STATE_CONFIGS.ON_LEFT_SPACE,
+          duration: SWITCH_STATE_CONFIGS.KNOB_SCALE_DURATION,
+        }).start(() => {
+          onChange(!on);
+        });
+      } else {
+        Animated.timing(rightSpace.current, {
+          toValue: SWITCH_STATE_CONFIGS.OFF_RIGHT_SPACE,
+          duration: SWITCH_STATE_CONFIGS.KNOB_SCALE_DURATION,
+        }).start(() => {
+          onChange(!on);
+        });
+      }
+      return;
+    }
+    if (toggleState) {
+      moveKnob(
+        [
+          {
+            animationProp: leftSpace.current,
+            toValue: SWITCH_STATE_CONFIGS.OFF_LEFT_SPACE,
+            duration: SWITCH_STATE_CONFIGS.KNOB_TRANSLATION_DURATION,
+          },
+          {
+            animationProp: rightSpace.current,
+            toValue: SWITCH_STATE_CONFIGS.OFF_RIGHT_SPACE,
+            duration: SWITCH_STATE_CONFIGS.KNOB_TRANSLATION_DURATION,
+          },
+        ],
+        () => {
+          setToggleState(!toggleState);
+          onChange(!toggleState);
+        },
+      );
+    } else {
+      moveKnob(
+        [
+          {
+            animationProp: leftSpace.current,
+            toValue: SWITCH_STATE_CONFIGS.ON_LEFT_SPACE,
+            duration: SWITCH_STATE_CONFIGS.KNOB_TRANSLATION_DURATION,
+          },
+          {
+            animationProp: rightSpace.current,
+            toValue: SWITCH_STATE_CONFIGS.ON_RIGHT_SPACE,
+            duration: SWITCH_STATE_CONFIGS.KNOB_TRANSLATION_DURATION,
+          },
+        ],
+        () => {
+          setToggleState(!toggleState);
+          onChange(!toggleState);
+        },
+      );
+    }
+  };
+
+  const interpolateContainerColor = leftSpace.current.interpolate({
+    inputRange: [SWITCH_STATE_CONFIGS.OFF_LEFT_SPACE, SWITCH_STATE_CONFIGS.ON_LEFT_SPACE],
+    outputRange: [theme.colors.shade[960], theme.colors.primary[800]],
+  });
+
+  const activeContainerColor = active
+    ? toggleState
+      ? theme.colors.primary[700]
+      : theme.colors.shade[950]
+    : null;
+
+  const disabledContainerColor = disabled
+    ? toggleState
+      ? theme.colors.primary[500]
+      : theme.colors.shade[930]
+    : null;
+
+  return (
+    <AnimatedContainer
+      disabled={disabled}
+      activeOpacity={1}
+      align={align}
+      size={size}
+      onPressIn={onPressIn}
+      onPressOut={onPressOut}
+      style={{
+        backgroundColor:
+          activeContainerColor || disabledContainerColor || interpolateContainerColor,
```

Comment:
```suggestion
        backgroundColor: interpolateContainerColor
```

You only need ```interpolateContainerColor```

Reply:
It is to account for disabled/active states. As discussed, resolving it.